packages/hyperion-autologging/src/ALSurfaceContext.ts

```diff
@@ -17,37 +20,55 @@ export type InitOptions =
   }>;
 
 
-type ALSurfaceContextValue = Readonly<{
+type ALSurfaceContextValue<
+  DataType extends ALFlowletDataType,
+  FlowletType extends Flowlet<DataType>
+> = Readonly<{
   surface: string,
+  flowlet: FlowletType | null,
```

Comment:
Just to make sure these values are always assigned together, you should define the type like the following
`{ surface: string; flowlet: FlowletType} | {surface: null; flowlet: null}`

TS is smart enough to not allow a broken combination. Then the code bellow becomes easier.